spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesServiceInstance.java

```diff
@@ -37,43 +34,49 @@
 
 	private static final String DSL = "//";
 
-	private static final String COLN = ":";
+	private static final String COLON = ":";
+
+	private String instanceId;
 
-	private final String instanceId;
+	private String serviceId;
 
-	private final String serviceId;
+	private String host;
 
-	private final EndpointAddress endpointAddress;
+	private int port;
 
-	private final EndpointPort endpointPort;
+	private URI uri;
 
-	private final Boolean secure;
+	private Boolean secure;
 
-	private final Map<String, String> metadata;
+	private Map<String, String> metadata;
 
 	/**
 	 * @param serviceId the id of the service.
-	 * @param endpointAddress the address where the service instance can be found.
-	 * @param endpointPort the port on which the service is running.
+	 * @param host the address where the service instance can be found.
+	 * @param port the port on which the service is running.
 	 * @param metadata a map containing metadata.
 	 * @param secure indicates whether or not the connection needs to be secure.
 	 * @deprecated - use other constructor
 	 */
 	@Deprecated
-	public KubernetesServiceInstance(String serviceId, EndpointAddress endpointAddress,
```

Comment:
Let's use this opportunity to remove the deprecated constructor.

Reply:
Done